### PR TITLE
Beginning C++23: Chapter 13

### PR DESCRIPTION
**Pull request by Peter-VanWeert**

### CODE REVIEW

File: Exercises/Modules/Chapter 13/Soln13_01/Box.cpp [hunk 1]
```diff
@@ -1,7 +1,6 @@
 module box;
 
-import <format>;
-import <algorithm>;    // For the min() and max() function templates
+import std;
 
 double Box::volume() const
 {
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_01/Box.cpp:6]: This comment is about line 20.
The op* is multiplying lenght, width and height with factor, but the exercise asks for "to produce a new object that has a height that is n times the original object" So, only the height should be multiplied.

File: Exercises/Modules/Chapter 13/Soln13_02/Box.cpp [hunk 2]
```diff
@@ -1,7 +1,6 @@
 module box;
 
-import <format>;
-import <algorithm>;    // For the min() and max() function templates
+import std;
 
 double Box::volume() const
 {
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_02/Box.cpp:6]: See previous comment. exercise 1 asked "to produce a new object that has a height that is n times the original object" so in both op*(double) and op*(double, box) only height should be multiplied.

File: Exercises/Modules/Chapter 13/Soln13_03/Box.cpp [hunk 3]
```diff
@@ -1,7 +1,6 @@
 module box;
 
-import <format>;
-import <algorithm>;    // For the min() and max() function templates
+import std;
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_03/Box.cpp:3]: Same comment as before about "to produce a new object that has a height that is n times the original object"

File: Exercises/Modules/Chapter 13/Soln13_04/Soln13_04.cpp [hunk 4]
```diff
@@ -1,6 +1,5 @@
-// Exercise 13-4
 // Adding support for equality operators that compare Boxes and volumes
-import <iostream>;
+import std;
 import box;
 
 /*
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_04/Soln13_04.cpp:5]: Comment for line 6: rephrase C++20?

File: Exercises/Modules/Chapter 13/Soln13_06/Box.cppm [hunk 5]
```diff
@@ -18,30 +16,31 @@ public:
...
   { 
     return volume() <=> otherVolume;
   }
 
   bool operator==(const Box& otherBox) const = default;
 
-  // Explicit type conversion operator (converts a Box to a Boolean; true if it has volume)
-  explicit operator bool() const  { return volume() != 0; }  
+  // Unary negation operator (!box is true if the Box has no volume)
+  bool operator!() const { return volume() == 0; }
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_06/Box.cppm:31]: Why did you add op!() one?
It all works with just the conversion operator below.

Comment (MarcGregoire) [reply Exercises/Modules/Chapter 13/Soln13_06/Box.cppm:31]: Ah, I just read the next exercise (7)... it's kinda on purpose

File: Exercises/Modules/Chapter 13/Soln13_08/Rational.cppm [hunk 6]
```diff
@@ -1,5 +1,5 @@
 export module rational;
-import <iostream>;
+import std;
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_08/Rational.cppm:3]: Several operators that define a copy are not using uniform init.

File: Exercises/Modules/Chapter 13/Soln13_08/Rational.cppm [hunk 7]
```diff
@@ -17,27 +17,27 @@ public:
...
-  bool operator==(const Rational& other)
+  bool operator==(const Rational& other) const
   {
     return m_numerator * other.m_denominator == other.m_numerator * m_denominator;
   }
-  bool operator==(double value)
+  bool operator==(double value) const
   {
     return static_cast<double>(*this) == value;
   }
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_08/Rational.cppm:43]: Typo towards end of file, it says myRationale, should be without ending e.

File: Exercises/Modules/Chapter 13/Soln13_10/Box.cppm [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  Box() = default;
+  Box(double length, double width, double height) 
+    : m_length{length}, m_width{width}, m_height{height} {};
+
+  double volume() const 
+  {
+    return m_length * m_width * m_height;
+  }
+
+  int compare(const Box& box) const
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_10/Box.cppm:17]: This is not the same implementation as Ex13_11 which the exercise text refers to.

File: Examples/Modules/Chapter 13/Ex13_05/Box.cpp [hunk 9]
```diff
@@ -1,7 +1,6 @@
 module box;
 
-import <format>;
-import <algorithm>;    // For the min() and max() function templates
+import std;
 
 double Box::volume() const
 {
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 13/Ex13_05/Box.cpp:6]: This comment is about line 12. Technically, it's not a "new" box. You're modifying the current box.

File: Examples/Modules/Chapter 13/Ex13_11/Box.cpp [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+double Box::volume() const
+{
+    return m_length * m_width * m_height;
+}
+
+// Overloaded += operator
+Box& Box::operator+=(const Box& aBox)
+{
+  // New object has larger length and width, and sum of heights
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 13/Ex13_11/Box.cpp:13]: Technically, it's not a "new" box. You're modifying the current box.

File: Exercises/Modules/Chapter 13/Soln13_04/Box.cppm [hunk 11]
```diff
@@ -18,11 +16,11 @@ public:
   double getWidth()  const { return m_width; }
   double getHeight() const { return m_height; }
 
-  std::partial_ordering operator<=>(const Box& otherBox) const 
+  auto operator<=>(const Box& otherBox) const
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_04/Box.cppm:19]: Up to now, it was always explicitly std::partial_ordering.

File: Exercises/Modules/Chapter 13/Soln13_04/Box.cppm [hunk 12]
```diff
@@ -18,11 +16,11 @@ public:
   double getWidth()  const { return m_width; }
   double getHeight() const { return m_height; }
 
-  std::partial_ordering operator<=>(const Box& otherBox) const 
+  auto operator<=>(const Box& otherBox) const 
   { 
     return volume() <=> otherBox.volume();
   }
-  std::partial_ordering operator<=>(double otherVolume) const 
+  auto operator<=>(double otherVolume) const
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_04/Box.cppm:23]: Up to now, it was always explicitly std::partial_ordering.

File: Exercises/Modules/Chapter 13/Soln13_06/Box.cppm [hunk 13]
```diff
@@ -18,30 +16,31 @@ public:
   double getWidth()  const { return m_width; }
   double getHeight() const { return m_height; }
 
-  std::partial_ordering operator<=>(const Box& otherBox) const 
+  auto operator<=>(const Box& otherBox) const
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_06/Box.cppm:19]: See previous comments.
Same for op<=>(double)

File: Exercises/Modules/Chapter 13/Soln13_09/Box.cpp [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+double Box::volume() const
+{
+    return m_length * m_width * m_height;
+}
+
+// Overloaded += operator
+Box& Box::operator+=(const Box& aBox)
+{
+  // New object has larger length and width, and sum of heights
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 13/Soln13_09/Box.cpp:13]: Technically, it's not a "new" box. You're modifying the current box.

**Review by MarcGregoire — CHANGES_REQUESTED** [review]

File: Examples/Modules/Chapter 13/Ex13_12B/Message.cppm [hunk 15]
```diff
@@ -1,6 +1,5 @@
-module;
-#include <cstring>  // For std::strlen() and std::strcpy()
 export module message;
+import std;
 
 export class Message
 {
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 13/Ex13_12B/Message.cppm:5]: This comment is about line 27:
return one.swap(other);
this should just be:
one.swap(other);

**Review by MarcGregoire — CHANGES_REQUESTED** [review]

File: Exercises/Modules/Chapter 13/Soln13_06/Box.cppm (hunk 5 above)
Comment (Peter-VanWeert) [reply Exercises/Modules/Chapter 13/Soln13_06/Box.cppm:31]: It wasn't on purpose originally. But now it is :-)

File: Exercises/Modules/Chapter 13/Soln13_04/Box.cppm (hunk 11 above)
Comment (Peter-VanWeert) [reply Exercises/Modules/Chapter 13/Soln13_04/Box.cppm:19]: No matter...

**Comment by Peter-VanWeert** [comment]
(Forgot to push my fixes before merging, but they're on main...)

**Pull request merged**